engine/src/main/scala/cromwell/engine/workflow/WorkflowManagerActor.scala

```diff
@@ -299,35 +299,38 @@ class WorkflowManagerActor(backend: Backend, config: Config)
     }
   }
 
-  private def assertCallFqnWellFormed(descriptor: WorkflowDescriptor, callFqn: FullyQualifiedName): Try[String] = {
+  private def assertCallFqnWellFormed(descriptor: WorkflowDescriptor, callFqn: FullyQualifiedName): Future[Unit] = {
     descriptor.namespace.resolve(callFqn) match {
-      case Some(c: Call) => Success(c.unqualifiedName)
-      case _ => Failure(new UnsupportedOperationException(s"Expected a fully qualified name to have at least two parts but got $callFqn"))
+      case None => Future.failed(new UnsupportedOperationException(
+        s"Expected a fully qualified name to have at least two parts but got $callFqn"))
+      case _ => Future.successful(())
     }
   }
 
   private def callStdoutStderr(workflowId: WorkflowId, callFqn: String): Future[AttemptedCallLogs] = {
 
-    def callKey(descriptor: WorkflowDescriptor, callName: String, key: ExecutionDatabaseKey) =
-      BackendCallKey(descriptor.namespace.workflow.findCallByName(callName).get, key.index, key.attempt)
+    def getCallLogs(workflowId: WorkflowId, keys: Seq[ExecutionDatabaseKey]): Future[AttemptedCallLogs] = {
+      val callLogFutures = keys map { key =>
+        globalDataAccess.getCallLogOutputs(workflowId, key) map { entries =>
+          key -> key.toCallLogs(entries)
+        }
+      }
 
-    def backendCallFromKey(workflow: WorkflowDescriptor, callName: String, key: ExecutionDatabaseKey) = {
-      BackendCallJobDescriptor(workflow, callKey(workflow, callName, key))
+      for {
+        results <- Future.sequence(callLogFutures)
+      } yield results.groupBy(_._1.index).toIndexedSeq.sortBy(_._1).map(_._2) map {
+        _.sortBy(_._1.attempt).map(_._2.get).toIndexedSeq
+      }
```

Comment:
Oh wow, any way this can be made more comprehensible?  I did do some cleanup on the PBE branch though it's possible that wasn't 100% correct.